audit/audit.cpp

```diff
@@ -113,7 +131,23 @@ audit::report_error(const std::string& short_name, const std::string& descriptio
 {
     this->recorded_errors.push_back(description);
     LOG(fatal) << boost::format("[%1%]: %2%") % short_name % description;
-    // TODO: Push to stats.d goes here
+    this->send_to_monitor(str(boost::format("%1%:1|c") % short_name));
+}
+
+void
+audit::send_to_monitor(const std::string& stat)
+{
+    if(!this->monitor_endpoint)
+    {
+        return;
+    }
+
+    LOG(debug) << boost::format("Sending stat '%1%' to monitor at %2%:%3%")
+                  % stat
+                  % this->monitor_endpoint->address().to_string()
+                  % this->monitor_endpoint->port();
+
+    this->socket->send_to(stat, *(this->monitor_endpoint));
```

Comment:
send_to is a blocking call. We try to avoid those as much as possible.
Please use async_send_to.

Reply:
You may need to wrap this in a strand.

Reply:
The whole thing is under mutexes held by the top level message handlers